IndicoV2/Submissions/SubmissionsClient.cs

```diff
@@ -25,47 +26,70 @@ public SubmissionsClient(IndicoClient indicoClient)
             _strawberryShakeClient = indicoClient.IndicoStrawberryShakeClient;
         }
 
-       
-        public Task<IEnumerable<int>> CreateAsync(int workflowId, IEnumerable<Stream> streams,
-            CancellationToken cancellationToken = default) =>
-            _legacy.CreateAsync(workflowId, streams, cancellationToken);
+        public async Task<IEnumerable<int>> CreateAsync(int workflowId, IEnumerable<Stream> streams, CancellationToken cancellationToken = default)
+        {
+            var uploadRequest = new UploadStream(_indicoClient)
+            {
+                Streams = streams.ToList()
+            };
+            var fileMetadata = await uploadRequest.Call();
+            var files = new List<object>();
+            foreach (JObject uploadMeta in fileMetadata)
+            {
+                var meta = new JObject
+                    {
+                        { "name", uploadMeta.Value<string>("name") },
+                        { "path", uploadMeta.Value<string>("path") },
+                        { "upload_type", uploadMeta.Value<string>("upload_type") }
+                    };
+
+                var file = new
+                {
+                    filename = uploadMeta.Value<string>("name"),
+                    filemeta = meta.ToString()
+                };
 
-        public async Task<IEnumerable<int>> CreateAsync(int workflowId, IEnumerable<(string Name, Stream Content)> filesToUpload,  CancellationToken cancellationToken = default, SubmissionResultsFileVersion? resultsFileVersion = null)
+                files.Add(file);
+            }
+            return await _strawberryShakeClient.Submissions().Create(workflowId, (IEnumerable<(string Name, string Meta)>)files, cancellationToken);
+        }
+
+        public async Task<IEnumerable<int>> CreateAsync(int workflowId, IEnumerable<(string Name, Stream Content)> filesToUpload, CancellationToken cancellationToken = default, SubmissionResultsFileVersion? resultsFileVersion = null)
         {
             var filesUploaded = await _indicoClient.Storage().UploadAsync(filesToUpload, cancellationToken);
             return await _strawberryShakeClient.Submissions().Create(workflowId, filesUploaded, cancellationToken, (SubmissionResultVersion?)resultsFileVersion);
         }
 
 
-     
+        [Obsolete("This is the Legacy version and will be deprecated. Please use CreateAsync instead.")]
         public Task<IEnumerable<int>> CreateAsyncLegacy(int workflowId, IEnumerable<string> paths,
```

Comment:
removed call to _legacy so that we can remove v1 while still marking as deprecated